lighthouse-cli/performance-experiment/experiment-database/database.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const rimraf = require('rimraf');
+const PerfXReportGenerator = require('../report/perf-x-report-generator');
+const assetSaver = require('../../../lighthouse-core/lib/asset-saver');
+
+class ExperimentDatabase {
+  constructor(url, config) {
+    this._url = url;
+    this._config = config;
+    this._defaultId = undefined;
+
+    this._root = fs.mkdtempSync(`${__dirname}/experiment-data-`);
+    this._timeStamps = {};
+  }
+
+  get url() {
+    return this._url;
+  }
+
+  get config() {
+    return this._config;
+  }
+
+  /*
+   * Set Default ID. When id is not provided to .getHTML() or .getFlags(), default ID will be used
+   * @param {!Object} lhFlags
+   * @param {!Object} lhResults
+   */
+  setDefaultId(id) {
+    this._defaultId = id;
+  }
+
+  /*
+   * Save experiment data
+   * @param {!Object} lhFlags
+   * @param {!Object} lhResults
+   */
+  saveData(lhFlags, lhResults) {
+    const id = assetSaver.getFilenamePrefix(lhResults);
+    this._timeStamps[id] = lhResults.generatedTime;
+
+    const dirPath = path.join(this._root, id);
+    fs.mkdirSync(dirPath);
+    fs.writeFileSync(path.join(dirPath, 'flags.json'), JSON.stringify(lhFlags));
+    fs.writeFileSync(path.join(dirPath, 'results.json'), JSON.stringify(lhResults));
+    return id;
+  }
+
+  /*
+   * Get report.html
+   * @param {?string} id When id is not provided, this._defaultId will be used
+   */
+  getHTML(id) {
+    id = id || this._defaultId;
```

Comment:
Ah. I see. it's a fallback solution so navigating to the web root will always give them a page to use.

Seems okay since we dont have a "please select a report on the left to view" index page yet.

I'd prefer this is handled at the router level rather than the database one. can we move it to the server?